auth/src/main/java/com/firebase/ui/auth/ui/idp/CredentialSignInHandler.java

```diff
@@ -121,8 +146,32 @@ public void onSuccess(@NonNull ProviderQueryResult result) {
                                         .setProvider(provider)
                                         .build(),
                                 mResponse
-                        ), mAccountLinkResultCode);
+                        ), mAccountLinkRequestCode);
+            }
+        }
+
+        private String getHighestPriorityProvider(List<String> providers) {
```

Comment:
I wrote this code a very long time ago, and this method seems a bit much to me. The simpler but less efficient solution would be to do this:
```java
if (providers.contains(GoogleAuthProvider.PROVIDER_ID)) {
    return GoogleAuthProvider.PROVIDER_ID;
} else if (providers.contains(FacebookAuthProvider.PROVIDER_ID)) {
    return FacebookAuthProvider.PROVIDER_ID;
} else if (providers.contains(TwitterAuthProvider.PROVIDER_ID)) {
    return TwitterAuthProvider.PROVIDER_ID;
} else {
    return providers.get(0);
}
```

Which one should I use?